presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -563,7 +563,9 @@ public static int compare(long leftRawLo, long leftRawHi, long rightRawLo, long
             return leftStrictlyNegative ? -1 : 1;
         }
         else {
-            return compareUnsigned(leftRawLo, leftRawHi, rightRawLo, rightRawHi) * (leftStrictlyNegative ? -1 : 1);
+            long leftHi = unpackUnsignedLong(leftRawHi);
+            long rightHi = unpackUnsignedLong(rightRawHi);
+            return compareUnsigned(leftRawLo, leftHi, rightRawLo, rightHi) * (leftStrictlyNegative ? -1 : 1);
```

Comment:
It is more intuitive when `compareUnsigned` method behaves in the same way as `Long.compareUnsigned` does. I would rather rename the `Slice` version to `compareIgnoreSign` or `compareAbsoluteValue`. From my perspective it is better reflects the purpose of this method.